### PR TITLE
[WIP] Changed Object browser logic to work with websockets

### LINKED ISSUE #943: Pagination in web UI
**Is your feature request related to a problem? Please describe.**

I know that it is not good practice to store a lot of object/subdirectories under bucket/directory but sometime it is unavoidable.
Currently I have about ~300 subfolders and ~1000 objects and it takes long time for Console's object browser or mc ls to preview them. I see that both of them try to load all "objects", but it is not always necessary so pagination can be good solution I think.

**Describe the solution you'd like**
Paginate list of items in buckets and dirs

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by bexsoft**
fixes #943

## What does this do?

This allows us to start streaming results to the objects list instead of waiting to retrieve all the objects before sending them to the client.

## How des it look?

Signed-off-by: Benjamin Perez <benjamin@bexsoft.net>

### CODE REVIEW

**Review by cniackz — DISMISSED** [review]
LGTM

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx [hunk 1]
```diff
@@ -542,192 +490,226 @@ const ListObjects = () => {
...
+        c.onclose = () => {
+          // reset start status
+          dispatch(setLoadingObjectsList(false));
+        };
+        c.onerror = (err) => {
+          console.error("Error", err);
+        };
+        return () => {
+          // close websocket on useEffect cleanup
+          //c.close(1000);
```
Comment (jinapurapu) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx:563]: Should this be commented out?

Comment (bexsoft) [reply portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx]: Good catch, this needs to be removed

Comment (bexsoft) [reply portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx]: Good catch.

This block is not necessary, I have removed it

**Review by cniackz — APPROVED** [review]
LGTM

**Review by jinapurapu — DISMISSED** [review]

**Review by dvaldivia — CHANGES_REQUESTED** [review]
There's a few problems:

After uploading a file, it shows up twice

Screen Shot 2022-10-07 at 1 50 29 PM

If the object is shown duplicate, even refreshing the list will show it twice.


Attempting to delete selected objects does nothing

**Comment by bexsoft** [comment]
> There's a few problems:
> 
> After uploading a file, it shows up twice
> 
> Screen Shot 2022-10-07 at 1 50 29 PM
> 
> If the object is shown duplicate, even refreshing the list will show it twice.  [  ](https://user-images.githubusercontent.com/18384552/194650662-a42bb3f1-1175-4412-9948-3d47f443c7a1.gif) [ ](https://user-images.githubusercontent.com/18384552/194650662-a42bb3f1-1175-4412-9948-3d47f443c7a1.gif)
> 
> Attempting to delete selected objects does nothing  [  ](https://user-images.githubusercontent.com/18384552/194650708-18ac6705-21e5-434e-9966-f9d38b3cf179.gif) [ ](https://user-images.githubusercontent.com/18384552/194650708-18ac6705-21e5-434e-9966-f9d38b3cf179.gif)

Cannot reproduce, Can you please provide some steps for this issue? Thank you!

https://user-images.githubusercontent.com/33497058/194670496-5c6e4dad-bb8c-4dd1-a1db-8ba1fcf7a38c.mp4

**Review by Alevsk — CHANGES_REQUESTED** [review]
[Screencast from 10-07-2022 05:20:49 PM.webm](https://user-images.githubusercontent.com/1795553/194678064-3f58bb28-3133-41d8-9fd3-dd31ce1d7c12.webm)

Testing this PR im not able to list objects inside a bucket, im using minio root credentials, if i use master branch then everything works fine

**Comment by bexsoft** [comment]
> [Screencast from 10-07-2022 05:20:49 PM.webm](https://user-images.githubusercontent.com/1795553/194678064-3f58bb28-3133-41d8-9fd3-dd31ce1d7c12.webm)
> 
> 
> 
> Testing this PR im not able to list objects inside a bucket, im using minio root credentials, if i use master branch then everything works fine

Please use 9090 or enable websockets support on 5005

**Review by Alevsk — APPROVED** [review]
Tested, works fine

**Review by prakashsvmx — APPROVED** [review]
Tested. Changes look good to me :+1:

**Pull request closed without merging**